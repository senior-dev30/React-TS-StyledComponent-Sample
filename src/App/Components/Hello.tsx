import React from 'react'
import styled from 'styled-components'

const Root = styled.div`
  padding: 50px 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

const ButtonsView = styled.div`
  width: 125px;
  display: flex;
  justify-content: space-between;
`

const Button = styled.button`
  height: 50px;
  width: 50px;
  background-color: #999;
  border-radius: 6px;
  border-color: #999;
  font-size: 24px;
`

const Greeting = styled.h1`
  color: #999;
  font-weight: bold;
`

export interface IProps {
  name: string
  enthusiasmLevel?: number
}

interface IState {
  enthusiasmLevel: number
}

export default class Hello extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props)

    if ((props.enthusiasmLevel || 0) <= 0) {
      throw new Error('You could be a little more enthusiastic. :D')
    }

    this.state = {
      enthusiasmLevel: props.enthusiasmLevel || 1,
    }
  }

  public onIncrement = () => {
    this.setState({ enthusiasmLevel: this.state.enthusiasmLevel + 1 })
  }
  public onDecrement = () => {
    const { enthusiasmLevel } = this.state
    if (enthusiasmLevel > 0) {
      this.setState({ enthusiasmLevel: enthusiasmLevel - 1 })
    }
  }
  public getExclamationMarks = (numChars: number) =>
    Array(numChars + 1).join('!')

  public render() {
    return (
      <Root>
        <Greeting>
          Hello{' '}
          {this.props.name +
            this.getExclamationMarks(this.state.enthusiasmLevel)}
        </Greeting>
        <ButtonsView>
          <Button
            type="button"
            onClick={this.onDecrement}
            aria-label="decrement"
          >
            -
          </Button>
          <Button
            type="button"
            onClick={this.onIncrement}
            aria-label="increment"
          >
            +
          </Button>
        </ButtonsView>
      </Root>
    )
  }
}
