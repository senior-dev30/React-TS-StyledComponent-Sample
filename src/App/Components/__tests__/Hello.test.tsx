import React from 'react'
import renderer from 'react-test-renderer'

import Hello from '../Hello'

test('renders correctly with defaults', () => {
  const tree = renderer
    .create(<Hello name="World" enthusiasmLevel={1} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
