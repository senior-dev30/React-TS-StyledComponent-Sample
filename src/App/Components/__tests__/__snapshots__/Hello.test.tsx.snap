// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<div
  className="sc-bdVaJa gTKXhV"
>
  <h1
    className="sc-bxivhb gLEdJP"
  >
    Hello
     
    World!
  </h1>
  <div
    className="sc-bwzfXH eQipRW"
  >
    <button
      aria-label="decrement"
      className="sc-htpNat iyPLpZ"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      aria-label="increment"
      className="sc-htpNat iyPLpZ"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
</div>
`;
